libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp

```diff
@@ -214,10 +221,24 @@ bool ESP8266WiFiAPClass::softAPConfig(IPAddress local_ip, IPAddress gateway, IPA
         return false;
     }
     struct ip_info info;
+	
     info.ip.addr = local_ip.v4();
     info.gw.addr = gateway.v4();
     info.netmask.addr = subnet.v4();
-
+	
+    uint32_t softap_ip = htonl(info.ip.addr);
+    uint32_t start_ip = htonl(dhcp_start.v4());
+    uint32_t end_ip = htonl(dhcp_end.v4());
+    softap_ip >>= 8;
+    if (( start_ip >> 8 != softap_ip ) || ( end_ip >> 8 != softap_ip )) {
+        DEBUG_WIFI("[APConfig] dhcp_start or dhcp_end isn't in range of local_ip Address!\n");

```

Comment:
This test is not right. v4 addresses need to be shifted by the number of zeroed bits in the netmask address. It is indeed 8 with 255.255.255.0 but since netmask is not hardcoded, "8" neither should be.
There are other attempts for a proper API for network calculations like #6196 .

Reply:
You are right. But this part of the code compares the start and the end IP with the local IP to be in the same range. And I copied it from the esp-dhcpserver.c file. for example :
If the local IP is 192.168.4.1, the start IP and end IP must be in range 192.168.4 .
The comparison with the netmask you are looking for is done elsewhere.

Reply:
I understand what you mean and you are absolutely right. But if the subnetmask is e.g 255.255.0.0 and the dhcp_start or dhcp_end is e.g 192.168.5.10, the "[APConfig] wifi_set_ip_info failed" error occurs in debug. Because of the following lines in the "esp-dhcpserver.c" file :

![Capture](https://user-images.githubusercontent.com/13591223/63928323-9d883300-ca64-11e9-8b10-ee4311bcc69f.JPG)

Reply:
Because you are right saying it is dependent on dhcp server,
I propose you test and exit with an error when `netmask != IPAddress(255,255,255,0)`.

Reply:
@d-a-v Thanks for your consideration.
Since the "DHCPS_MAX_LEASE" is limited to 100, we don't need more than 254 IPs in a range. And if we were to limit the "subnet mask" to 255.255.255.0, why would there be an option to set the "subnet mask"? And wouldn't it be better to implement "softAPConfig" without it as follows?
```cpp
bool ESP8266WiFiAPClass::softAPConfig(IPAddress local_ip, IPAddress gateway,  IPAddress dhcp_start, IPAddress dhcp_end) {

IPAddress subnet = (255, 255, 255, 0);
info.netmask.addr = subnet.v4();
```